documentation.md

```diff
@@ -0,0 +1,245 @@
+# Cyberwatch API Toolbox documentation
+
+## Available methods
+
+#### Ping
+
+Sends a GET request to `/api/v2/ping` to validate that the API is working.
+
+###### Usage example and expected result:
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).ping()
+INFO:root:OK
+True
+```
+
+#### Servers
+
+Send a GET request to `/api/v2/servers` to retrieve the list of all servers.
+
+###### Usage example and expected result:
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).servers()
+[<cbw_api_toolbox.cbw_objects.cbw_server.CBWServer ...]
+```
+
+#### Server
+
+Send a GET request to `/api/v2/servers/{SERVER_ID}` to retrieve the information of SERVER_ID.
+
+##### Usage example and expected result:
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).server(SERVER_ID)
+[<cbw_api_toolbox.cbw_objects.cbw_server.CBWServer]
+```
+
+#### Update server
+
+Send a PATCH request  `/api/v2/servers/{SERVER_ID}` to update the information of SERVER_ID.
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).update_server(SERVER_ID, INFO)
+>>>True
+```
+
+#### Delete server
+
+Send a DELETE request `/api/v2/servers/{SERVER_ID}` to delete SERVER_ID.
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).delete_server(SERVER_ID)
+>>>True
+```
+
+#### Remote accesses
+
+Send a GET request `/api/v2/remote_accesses` to retrieve the list of all remote accesses.
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).remote_accesses()
+[<cbw_api_toolbox.cbw_objects.cbw_remote_access.CBWRemoteAccess...]
+```
+
+#### Remote access
+
+Send a GET request `/api/v2/remote_accesses/{REMOTE_ACCESS_ID}` to retrieve the information of REMOTE_ACCESS_ID.
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).remote_access(REMOTE_ACCESS_ID)
+[<cbw_api_toolbox.cbw_objects.cbw_remote_access.CBWRemoteAccess]
+```
+
+#### Create remote access
+
+Send a POST request `/api/v2/remote_accesses` to create a remote access
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).create_remote_access(INFO)
+INFO:root:remote access successfully created {ADDRESS}
+True
+```
+
+#### Update remote access
+
+Send a PATCH request `/api/v2/remote_accesses/{REMOTE_ACCESS_ID}` to update the information of REMOTE_ACCESS_ID
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).update_remote_access(REMOTE_ACCESS_ID, INFO)
+True
+```
+
+#### Delete remote access
+
+Send a DELETE request `/api/v2/remote_accesses/{REMOTE_ACCESS_ID}` to delete REMOTE_ACCESS_ID
+
+```python
+>>> CBWApi(URL, API_KEY, SECRET_KEY).delete_remote_access(REMOTE_ACCESS_ID)
+True
+```
+
+## Available objects and their attributes
+
+#### Server
```

Comment:
For objects and their attributes, I believe :
- We should write the name of the object as it is in the code (eg. `Server` => `CBWServer`) as it is clearer and avoids confusion with methods
- The attributes listed in the documentation should match exactly the object's attributes (doesn't seem to be the case right now, at least with the CBWCve object here... see comments below about the object Server)

I didn't review and comment on every objects, but these changes should be applied to all documented objects